subsys/bluetooth/audio/ascs.c

```diff
@@ -51,49 +51,19 @@ BUILD_ASSERT(CONFIG_BT_ASCS_MAX_ACTIVE_ASES <= MAX(MAX_ASES_SESSIONS,
 	(_id > CONFIG_BT_ASCS_ASE_SNK_COUNT ? BT_UUID_ASCS_ASE_SRC : BT_UUID_ASCS_ASE_SNK)
 #define ASE_COUNT (CONFIG_BT_ASCS_ASE_SNK_COUNT + CONFIG_BT_ASCS_ASE_SRC_COUNT)
 
-struct bt_ascs_ase {
-	struct bt_ascs *ascs;
+static struct bt_ascs_ase {
+	struct bt_conn *conn;
 	struct bt_bap_ep ep;
 	const struct bt_gatt_attr *attr;
-	sys_snode_t node;
 	struct k_work_delayable disconnect_work;
-};
-
-struct bt_ascs {
-	struct bt_conn *conn;
-	sys_slist_t ases;
-};
```

Comment:
It does feel a bit weird that we don't have an ASCS struct, as current (before this PR) structure with an ASCS per connection pointer, and then 0 or more ASEs per ASCS feels like a more logical structure, and better adheres to the way that the ASCS specification is written. 

However, based on the changes in this PR, I agree that it seems that there is no technical advantage of the above structure, and I do not see any alternatives that is using less resources that what this PR implements.

Reply:
TL;DR
I've checked hap_ha sample and the result is as follows
It saves 40bytes of RAM and 312bytes of ROM.
Bear in mind that after the changes, the ASCS size is independent of `CONFIG_BT_MAX_CONN`.

`CONFIG_BT_MAX_CONN=1`:
Before:
![image](https://user-images.githubusercontent.com/10126877/235123975-b3716c2f-e262-4b6c-99a0-92b86fc18077.png)
RAM
![image](https://user-images.githubusercontent.com/10126877/235124259-020285bb-e493-4fdb-93dd-89e62e3d263f.png)
ROM
![image](https://user-images.githubusercontent.com/10126877/235124574-9c6a58c7-5f3b-4993-8455-d339fd574f76.png)

After:
![image](https://user-images.githubusercontent.com/10126877/235124892-e6f3d274-df8d-4ea7-b9cf-1dbfb0242134.png)
RAM
![image](https://user-images.githubusercontent.com/10126877/235124997-d637e4a7-5fa6-4627-8d16-580af9f5f08e.png)
ROM
![image](https://user-images.githubusercontent.com/10126877/235125183-f6ad5fd9-afff-461c-bcb3-43bceef1fa87.png)

Currently, setting `CONFIG_BT_MAX_CONN=2` affects the memory usage as follows:
![image](https://user-images.githubusercontent.com/10126877/235126867-e46e2249-eb62-423b-b19b-4e53d377285d.png)
![image](https://user-images.githubusercontent.com/10126877/235127128-8d2e5ee2-6af3-4642-91c1-f13e9af821eb.png)
As you can see, increasing the number of connections by one increased RAM usage by 12bytes more

Reply:
Very nice; thanks for the report :)